tests/unit/providers/test_aws_provider.py

```
@@ -790,3 +795,353 @@ def test_convert_tool_spec_empty_description() -> None:
+            CompletionParams(
+                model_id="us.anthropic.claude-haiku-4-5-20251001-v1:0",
+                messages=[{"role": "user", "content": "hi"}],
+                response_format=_City,
+                reasoning_effort="high",
+            ),
+            {},
+        )
+
+
+@pytest.mark.parametrize("reasoning_effort", ["none", "auto", None])
+def test_convert_params_response_format_allows_disabled_reasoning(reasoning_effort: Any) -> None:
```

Ruff (0.16.1)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `reasoning_effort`

(ANN401)